### PR TITLE
Fix several bugs running non-scanner aux modules

### LINKED ISSUE #11475: auxiliary/server/tftp  ERROR: Auxiliary failed: option RHOSTS failed to validate.
Expected behavior: Server starts after setting SRVHOST and executing "run".

Observed behavior: When setting appropriate options (this has always worked in the past) the module errors out with the following after execution:

[-] Auxiliary failed: option RHOSTS failed to validate.

Has something changed since previous MSF version? Not sure why it wants RHOSTS, especially given there is no option to set it in the module.

### PR DESCRIPTION
**Pull request by busterb**
This fixes a couple of bugs in #11176:

 - RHOSTS is not always used in Aux modules, don't enforce it unless it's used.
 - Be sure to pass the action to the run stub, since it's not a standard option.

## Verification

- [x] Start `msfconsole`
- [x] `use auxiliary/server/tftp`
- [x] `run`
- [x] **Verify** the module starts as expected

```
msf5 auxiliary(server/tftp) > run
[*] Auxiliary module running as background job 0.
msf5 auxiliary(server/tftp) > 
[*] Starting TFTP server on 0.0.0.0:6969...
[*] Files will be served from /var/folders/7q/jwhmljqn6nx8lnqft8nf5w2myy7k2k/T
[*] Uploaded files will be saved in /var/folders/7q/jwhmljqn6nx8lnqft8nf5w2myy7k2k/T
```

Fixes #11475 #11448 #11474 #11480

### CODE REVIEW

File: lib/msf/ui/console/command_dispatcher/auxiliary.rb [hunk 1]
```diff
@@ -123,23 +123,25 @@ def cmd_run(*args)
       jobify = true
     end
 
-    rhosts_range = Rex::Socket::RangeWalker.new(mod.datastore['RHOSTS'])
-    unless rhosts_range && rhosts_range.length
-      print_error("Auxiliary failed: option RHOSTS failed to validate.")
-      return false
+    unless mod.datastore['RHOSTS'].nil?
```
Comment (Green-m) [inline lib/msf/ui/console/command_dispatcher/auxiliary.rb]: How about `if mod.datastore['RHOSTS']` ?

**Comment by Green-m** [comment]
Sorry for the bug, fix a old bug and make serveral new bugs....

File: lib/msf/ui/console/command_dispatcher/auxiliary.rb (hunk 1 above)
Comment (busterb) [reply lib/msf/ui/console/command_dispatcher/auxiliary.rb]: Sure, tested and it worked fine.

**Comment by busterb** [comment]
It happens. We're looking at how to automate testing of these module types next as part of the sanity test too.

**Comment by bcoles** [comment]
> 
> 
> Sorry for the bug, fix a old bug and make serveral new bugs....

We call that progress. Two steps forward, one step back. Can't spell `progress` without `regr[e]ss`.

**Comment by bcoles** [comment]
Preliminary tests indicate this patch resolves the issue as described.

```
# wget 'https://github.com/rapid7/metasploit-framework/raw/287667d4f70375750e40c4ff0ea41762f1f48983/lib/msf/ui/console/command_dispatcher/auxiliary.rb'
--2019-02-26 05:51:29--  https://github.com/rapid7/metasploit-framework/raw/287667d4f70375750e40c4ff0ea41762f1f48983/lib/msf/ui/console/command_dispatcher/auxiliary.rb
Resolving github.com (github.com)... 192.30.255.113, 192.30.255.112
Connecting to github.com (github.com)|192.30.255.113|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://raw.githubusercontent.com/rapid7/metasploit-framework/287667d4f70375750e40c4ff0ea41762f1f48983/lib/msf/ui/console/command_dispatcher/auxiliary.rb [following]
--2019-02-26 05:51:30--  https://raw.githubusercontent.com/rapid7/metasploit-framework/287667d4f70375750e40c4ff0ea41762f1f48983/lib/msf/ui/console/command_dispatcher/auxiliary.rb
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.80.133
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.80.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 5391 (5.3K) [text/plain]
Saving to: ‘auxiliary.rb’

auxiliary.rb                100%[===========================================>]   5.26K  --.-KB/s    in 0s      

2019-02-26 05:51:31 (14.1 MB/s) - ‘auxiliary.rb’ saved [5391/5391]

# mv auxiliary.rb lib/msf/ui/console/command_dispatcher/auxiliary.rb
# ./msfconsole 
[-] ***rting the Metasploit Framework console...\
[-] * WARNING: No database support: No database YAML file
[-] ***
                                                  
                          ########                  #
                      #################            #
                   ######################         #
                  #########################      #
                ############################
               ##############################
               ###############################
              ###############################
              ##############################
                              #    ########   #
                 ##        ###        ####   ##
                                      ###   ###
                                    ####   ###
               ####          ##########   ####
               #######################   ####
                 ####################   ####
                  ##################  ####
                    ############      ##
                       ########        ###
                      #########        #####
                    ############      ######
                   ########      #########
                     #####       ########
                       ###       #########
                      ######    ############
                     #######################
                     #   #   ###  #   #   ##
                     ########################
                      ##     ##   ##     ##
                            https://metasploit.com


       =[ metasploit v5.0.9-dev-d78a9978e0                ]
+ -- --=[ 1859 exploits - 1057 auxiliary - 327 post       ]
+ -- --=[ 546 payloads - 44 encoders - 10 nops            ]
+ -- --=[ 2 evasion                                       ]

msf5 > use auxiliary/server/socks4a
msf5 auxiliary(server/socks4a) > run -j
[*] Auxiliary module running as background job 0.

[*] Starting the socks4a proxy server
msf5 auxiliary(server/socks4a) > lsof -i :1080
[*] exec: lsof -i :1080

COMMAND  PID USER   FD   TYPE DEVICE SIZE/OFF NODE NAME
ruby    2584 root   10u  IPv4 378554      0t0  TCP *:socks (LISTEN)
msf5 auxiliary(server/socks4a) > use auxiliary/scanner/ssh/ssh_enumusers
msf5 auxiliary(scanner/ssh/ssh_enumusers) > set rhosts 127.0.0.1
rhosts => 127.0.0.1
msf5 auxiliary(scanner/ssh/ssh_enumusers) > set USER_FILE /usr/share/metasploit-framework/data/wordlists/unix_users.txt
USER_FILE => /usr/share/metasploit-framework/data/wordlists/unix_users.txt
msf5 auxiliary(scanner/ssh/ssh_enumusers) > run

[*] 127.0.0.1:22 - SSH - Using malformed packet technique
[*] 127.0.0.1:22 - SSH - Starting scan
^C[*] Caught interrupt from the console...
[*] Auxiliary module execution completed
msf5 auxiliary(scanner/ssh/ssh_enumusers) > # ^ not a real ssh server
^CInterrupt: use the 'exit' command to quit
msf5 auxiliary(scanner/ssh/ssh_enumusers) > use auxiliary/server/capture/smb
msf5 auxiliary(server/capture/smb) > run -jz
[*] Auxiliary module running as background job 1.

[*] Started service listener on 0.0.0.0:445 
[*] Server started.
msf5 auxiliary(server/capture/smb) > lsof -i :445
[*] exec: lsof -i :445

COMMAND  PID USER   FD   TYPE DEVICE SIZE/OFF NODE NAME
ruby    2584 root   12u  IPv4 383166      0t0  TCP *:microsoft-ds (LISTEN)
msf5 auxiliary(server/capture/smb) > use auxiliary/server/tftp
msf5 auxiliary(server/tftp) > run -jz
[*] Auxiliary module running as background job 2.

[*] Starting TFTP server on 0.0.0.0:69...
[*] Files will be served from /tmp
[*] Uploaded files will be saved in /tmp
msf5 auxiliary(server/tftp) > lsof -i :69
[*] exec: lsof -i :69

COMMAND  PID USER   FD   TYPE DEVICE SIZE/OFF NODE NAME
ruby    2584 root   14u  IPv4 387364      0t0  UDP *:tftp 
msf5 auxiliary(server/tftp) > 
```

**Review by bcoles — APPROVED** [review]

**Comment by benichmt1** [comment]
fixed it for me, cheers

**Comment by busterb** [comment]
Thanks @bcoles @benichmt1 @Green-m

**Comment by busterb** [comment]
## Release Notes

This fixes a issue running passive and action-oriented auxiliary modules.

**Pull request merged**